pkgs/by-name/cm/cmake/package.nix

```diff
@@ -178,10 +181,51 @@ stdenv.mkDerivation (finalAttrs: {
 
   doCheck = false; # fails
 
-  passthru.updateScript = gitUpdater {
-    url = "https://gitlab.kitware.com/cmake/cmake.git";
-    rev-prefix = "v";
-    ignoredVersions = "-"; # -rc1 and friends
+  passthru = {
+    tests =
+      let
+        shellcheckBashSourceFile =
```

Comment:
This function deserves its own file. It can be useful for other setup hooks.

Reply:
The tests are merely shell-checking? Why?

In the worst hypothesis this files will never need a change before Cmake 4.

Reply:
> The tests are merely shell-checking? Why?
> 
> In the worst hypothesis this files will never need a change before Cmake 4.

There was no test at all.

This PR merely fixes the shell script. All the CMake logic stays unchanged, including the CMakeList.txt patching, name extraction, and the result CMake command being executed.

Should I add some instrumentation tests to ensure that the script handles the specified flags correctly?

To be fair, the ShellCheck tests are not trivial as it seems, considering the vast quality improvement of `cmake/setup-hook.sh` just to pass the check.

Reply:
Well, why are you adding these tests?

Usually what I expect from a test in `passthru.tests` is that it fails (with a loud scream) when an update is going to break something.

The typical example is `live555`: it can possibly compile and run fine, but it is a dependency of `vlc`. 
If the API of live555 changes and vlc does not catch this change, then it is not recommended to merge a new live555.
Then we added vlc as passthru.tests of live555.

On the other hand, the tests you added here do nothing besides linting our own code.
It merely checks if we did a good job writing our own shell script utilities.

And it is unlikely these scripts will ever change - maybe it will happen only when a new release of cmake exposes new functionalities (and/or removes some).

That being said, these tests do not look so useful. When a new release of cmake arrives, my last concern is if the setup hook code still shell-checks.